sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -963,387 +981,186 @@ class MultiFileParquetPartitionReader(
-          // use a single buffer and slice it up for different files if we need
-          val outLocal = hmb.slice(offset, fileBlockSize)
-          // copy the blocks for each file in parallel using background threads
-          tasks.add(MultiFileThreadPoolFactory.submitToThreadPool(
-            new ParquetCopyBlocksRunner(file, outLocal, blocks, offset),
-            numThreads))
-          offset += fileBlockSize
-        }
-
-        for (future <- tasks.asScala) {
-          val (blocks, bytesRead) = future.get()
-          allOutputBlocks ++= blocks
-          TrampolineUtil.incBytesRead(inputMetrics, bytesRead)
-        }
+  override def calculateEstimatedBlocksOutputSize(currentChunkedBlocks: Seq[DataBlockBase],
+      schema: SchemaBase): Long = {
+    calculateParquetOutputSize(currentChunkedBlocks, schema, true)
+  }
 
-        // The footer size can change vs the initial estimated because we are combining more blocks
-        //  and offsets are larger, check to make sure we allocated enough memory before writing.
-        // Not sure how expensive this is, we could throw exception instead if the written
-        // size comes out > then the estimated size.
-        val actualFooterSize = calculateParquetFooterSize(allOutputBlocks, clippedSchema)
-        // 4 + 4 is for writing size and the ending PARQUET_MAGIC.
-        val bufferSizeReq = offset + actualFooterSize + 4 + 4
-        out.close()
-        val totalBufferSize = if (bufferSizeReq > initTotalSize) {
-          logWarning(s"The original estimated size $initTotalSize is to small, " +
-            s"reallocing and copying data to bigger buffer size: $bufferSizeReq")
-          val prevhmb = hmb
-          val in = new HostMemoryInputStream(prevhmb, offset)
-          hmb = reallocHostBufferAndCopy(in, bufferSizeReq)
-          prevhmb.close()
-          bufferSizeReq
-        } else {
-          initTotalSize
-        }
-        val lenLeft = totalBufferSize - offset
-        val finalizehmb = hmb.slice(offset, lenLeft)
-        val footerOut = new HostMemoryOutputStream(finalizehmb)
-        writeFooter(footerOut, allOutputBlocks, clippedSchema)
-        BytesUtils.writeIntLittleEndian(footerOut, footerOut.getPos.toInt)
-        footerOut.write(ParquetPartitionReader.PARQUET_MAGIC)
-        val amountWritten = offset + footerOut.getPos
-        footerOut.close()
-        // triple check we didn't go over memory
-        if (amountWritten > totalBufferSize) {
-           throw new QueryExecutionException(s"Calculated buffer size $totalBufferSize is to " +
-            s"small, actual written: ${amountWritten}")
-        }
-        if (finalizehmb != null) {
-          finalizehmb.close()
-        }
-        (hmb, amountWritten)
-      }
-    }
+  override def getThreadPool(numThreads: Int): ThreadPoolExecutor = {
+    ParquetMultiFileThreadPoolFactory.getThreadPool(getFileFormatShortName, numThreads)
```

Comment:
so it looks like this is changing to share a threadpool between coalescing and cloud, I assume that means the other can be removed. (MultiFileThreadPoolFactory).